src/main/java/io/mystore/cart/entity/ShoppingCart.java

```diff
@@ -342,8 +342,8 @@ static CartEntity.CartState updateState(CartEntity.CartState state, CartEntity.C
   static List<CartEntity.LineItem> updateLineItemQuantity(CartEntity.CartState state, CartEntity.ItemAdded event) {
     return state.getLineItemsList().stream()
         .map(lineItem -> {
-          if (lineItem.getSkuId().equals(lineItem.getSkuId())) {
-            return lineItem.toBuilder()
+          if (lineItem.getSkuId().equals(event.getLineItem().getSkuId())) {
```

Comment:
This was a bug. It was returns true and updates all times with the new quantity.

(also below in line 359)